flake/dev/checks.nix

```
@@ -0,0 +1,47 @@
+{ inputs, ... }:
+{
+  perSystem =
+    {
+      lib,
+      system,
+      config,
+      pkgs,
+      ...
+    }:
+    {
+      checks = lib.mkMerge [
+        config.packages
+        {
+          # TODO: consider using https://flake.parts/options/git-hooks-nix.html
+          git-hooks = inputs.git-hooks.lib.${system}.run {
+            hooks = {
+              deadnix.enable = true;
+              editorconfig-checker.enable = true;
+              hlint.enable = true;
+
+              treefmt = {
+                enable = true;
+                package = config.formatter;
+              };
+
+              statix.enable = true;
+              typos = {
+                enable = true;
+                settings.configuration = ''
+                  [default.extend-identifiers]
+                  MrSom3body="MrSom3body"
+                '';
+              };
+              yamllint.enable = true;
+            };
+
+            src = ./.;
```

Since this config is moved into a subdir, `./.` is no longer correct:

```suggestion
            src = self.outPath;
```

or equivialent

```suggestion
            src = "${self}";
```

NOTE: switching to the flake-module resolves this as it sets a default, see patch below: